### PR TITLE
AWS: set lastEvaluatedKey for listTables in DynamoDb Catalog

### PR DESCRIPTION
**Pull request by munendrasn**
While working on issue #6763 found that lastEvaluatedKey was not set for subsequent calls in listTables()

@jackye1995 FYI

### CODE REVIEW

File: aws/src/main/java/org/apache/iceberg/aws/dynamodb/DynamoDbCatalog.java [hunk 1]
```diff
@@ -329,7 +329,7 @@ public boolean removeProperties(Namespace namespace, Set<String> properties)
   @Override
   public List<TableIdentifier> listTables(Namespace namespace) {
     List<TableIdentifier> identifiers = Lists.newArrayList();
-    Map<String, AttributeValue> lastEvaluatedKey;
+    Map<String, AttributeValue> lastEvaluatedKey = null;
```
Comment (jackye1995) [inline aws/src/main/java/org/apache/iceberg/aws/dynamodb/DynamoDbCatalog.java:332]: I don't think this change is needed? if not assigned it's default to null

Comment (munendrasn) [reply aws/src/main/java/org/apache/iceberg/aws/dynamodb/DynamoDbCatalog.java:332]: For local variables, if used without initialisation, compilation fails (using Jdk8). `error: variable lastEvaluatedKey might not have been initialized`

Comment (jackye1995) [reply aws/src/main/java/org/apache/iceberg/aws/dynamodb/DynamoDbCatalog.java:332]: I see, then it looks good to me

**Comment by jackye1995** [comment]
Could you add an integration test about this?

**Review by mohitgargk — APPROVED** [review]

**Comment by munendrasn** [comment]
I thought of updating this [test](https://github.com/apache/iceberg/blob/ef7e20e12eae2638015ff91b3356eb2f32f601cc/aws/src/integration/java/org/apache/iceberg/aws/dynamodb/TestDynamoDbCatalog.java#L213) to create more tables, and validate the change.

As DynamoDb Query reads max 1MB of data (when limit is not specified), need to create large number of tables (~10K with current setup) to reproduce the issue. Please let me know if I should go ahead, and update  number of tables to higher number in the tests

Locally,  I could verify the fix by adding `limit` parameter to the query request and then calling listTables on DynamoDb with & w/o the fix

**Review by jackye1995 — APPROVED** [review]
I see, that's probably why the bug has never been reported. In that case, I am good for now. 

We are working on some refactoring of the AWS integration tests at the moment, and can do something about this case afterwards for scale testing. We can let you know when you can add more tests for the DynamoDB catalog to ensure its correctness.

**Comment by amogh-jahagirdar** [comment]
+1 for adding some more AWS integration tests specifically focused on scale testing

**Review by amogh-jahagirdar — APPROVED** [review]
This PR looks good to me, since it's more involved to have automated tests for this we can table it for another effort to have better scale tests for our AWS integrations.

**Review by yyanyy — APPROVED** [review]
thanks for fixing this!

**Comment by jackye1995** [comment]
Thanks for the fix! @munendrasn

**Comment by munendrasn** [comment]
Thank you all for the review 🙂

**Pull request merged**